popup.js

```diff
@@ -1,111 +1,157 @@
     search(document.getElementById("textarea").value);
 });
 
-
 //donwload list as an html file
-save.addEventListener('click', function(){
-  downloadToFile();
+save.addEventListener("click", function () {
+    downloadToFile();
 });
 
 //clear list
-clear.addEventListener('click', function(){
-  clearList();
+clear.addEventListener("click", function () {
+    clearList();
 });
 
-
-//tile buttons search text
-document.getElementsByClassName("searchTile").addEventListener('click', function(event){
-  searchTile(event);
+// retrieves items stored in background.js
+chrome.runtime.onMessage.addListener(function(request, sender, sendResponse){
+    if (request.retrieve == "retrieve") {
+        console.log("received retrievable items")
+        request.itemsToAdd.forEach(function(item){
+            addItemToList(item);
+        })
+    }
 });
 
-//tile buttons delete
-document.getElementsByClassName("deleteTile").addEventListener('click', function(event){
-deleteTile(event);
-});
+function addItemToList(item) {
+    // Button 1
+    var button1 = document.createElement("button");
+    button1.classList.add("tooltip", "tilebutton", "searchTile");
+    var spans = document.createElement("span");
+    spans.classList.add("tooltiptext");
+    spans.appendChild(document.createTextNode("Search Note"));
+    button1.appendChild(spans);
+
+    // Button 2
+    var button2 = document.createElement("button");
+    button2.classList.add("tooltip", "tilebutton", "deleteTile");
+    var spans = document.createElement("span");
+    spans.classList.add("tooltiptext");
+    spans.appendChild(document.createTextNode("Delete Note"));
+    button2.appendChild(spans);
+
+    var texts = document.createElement("p");
+    texts.contentEditable = true;
+    texts.classList.add("editable");
+    texts.innerHTML = item;
+    var entry = document.createElement("li");
+    entry.appendChild(texts);
+    entry.setAttribute("id", "item" + index);
+    var d = new Date();
+    var date =
+        "<div id='date'>" +
+        d.getHours() +
+        ":" +
+        d.getMinutes() +
+        " " +
+        d.getDate() +
+        "/" +
+        d.getMonth() +
+        "/" +
+        d.getFullYear() +
+        "</div>";
+    var dates = document.createElement("p");
+    dates.innerHTML = date;
+    entry.appendChild(dates);
+    // entry.appendChild(document.createTextNode(texts));
+
+    // Button 1
+    var searchimg = document.createElement("img");
+    searchimg.src = "/images/search.png";
+    entry.appendChild(button1);
+    button1.appendChild(searchimg);
+
+    // Button 2
+    var deleteimg = document.createElement("img");
+    deleteimg.src = "/images/delete.png";
+    entry.appendChild(button2);
+    button2.appendChild(deleteimg);
+
+    if (item != "") list.appendChild(entry);
+    document.getElementById("textarea").value = "";
+    localStorage.setItem("listItems", list.innerHTML);
+    index += 1;
+
+    assignButton();   
```

Comment:
This is a _big_ function with lots of repeated code.

Think about breaking it up into smaller functions. 

Further reading: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself